src/test/java/com/akademia/projectplanner/service/impl/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,100 @@
+package com.akademia.projectplanner.service.impl;
+
+import com.akademia.projectplanner.dto.UserDto;
+import com.akademia.projectplanner.entity.UserEntity;
+import com.akademia.projectplanner.exception.AuthenticationException;
+import com.akademia.projectplanner.mapper.TaskMapper;
+import com.akademia.projectplanner.mapper.UserMapper;
+import com.akademia.projectplanner.repository.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+class RegistrationServiceImplTest {
+
+  @Mock private UserRepository userRepository;
+  @Mock private UserMapper userMapper;
+  @InjectMocks private TaskMapper taskMapper;
+  @InjectMocks private RegistrationServiceImpl registrationService;
+  private AutoCloseable autoCloseable;
+  private UserDto userDto;
+
+  @BeforeEach
+  void setUp() {
+    autoCloseable = MockitoAnnotations.openMocks(this);
+    userDto = new UserDto();
+    userDto.setId(1L);
+    userDto.setEmail("email");
+    userDto.setName("name");
+    userDto.setPassword("pass");
+    userDto.setPasswordRepeated("pass");
+  }
+
+  @Test
+  void shouldRegisterSuccessfully() {
+    // given
+    UserEntity userEntity = new UserEntity();
+    Mockito.when(userRepository.existsByEmail(userDto.getEmail())).thenReturn(false);
+    Mockito.when(userMapper.toUserEntity(userDto)).thenReturn(userEntity);
+    // when
+
+    registrationService.register(userDto);
+    // then
+    verify(userRepository, times(1)).save(userEntity);
+  }
+
+  @Test
+  void shouldThrowAuthenticationExceptionWhenEmailExists() {
```

Comment:
uwaga / hint / ciekawostka
Ogólnie w testach rzucających Exceptiony powinniśmy dawać w sygnaturze throws czyli np:
@Test
void nazwaTestu() throws MojException {
...
}

Jest to informacja dla kompilatora że tutaj wydarzy się coś niedobrego, podobnie jak przy normalnej metodzie w której też dajemy throws. Alternatywnie dla throws w sygnaturze cała zawartość testu powinna być zawarta w try catchu  poza samą assercją (ale nigdy tego nie widziałem osobiście - czyściej jest dawać throwsa w sygnaturze).
Junit sobie i tak z tym radzi  więc wszystko działa ale warto wiedzieć na przyszłość :)

https://blog.aspiresys.pl/technology/different-ways-of-testing-exceptions-in-java-and-junit/